sway/input/text_input.c

```diff
@@ -255,6 +267,240 @@ static void relay_handle_text_input(struct wl_listener *listener,
 	sway_text_input_create(relay, wlr_text_input);
 }
 
+static struct sway_layer_surface *find_layer_by_surface(
+		struct wlr_layer_surface_v1 *layer) {
+	for (int i = 0; i < root->outputs->length; ++i) {
+		struct sway_output *output = root->outputs->items[i];
+		// For now we'll only check the overlay and top layer
+		struct wlr_scene_node *node;
+		wl_list_for_each (node, &output->layers.shell_top->children, link) {
```

Comment:
I would extract this loop into a function so that you don't have to open code the loop twice.